### ISSUE TITLE
New England Soundproofing: Contact Form captcha not working?

### ISSUE THREAD

**Issue by KathJohnson** [meetinghouse/cms#253]
I spoke with Steve Drago at NES today, and he asked me to send him an email from the contact us page form as a test.

I was unable to do so because the security image won't load, so I can't enter the captcha content.

By the way, he would like to launch his site live on Tuesday morning, May 17, barring any delays that may come up. Thanks.

![screenshot 2016-05-10 at 11 18 47 am](https://cloud.githubusercontent.com/assets/5922040/15151871/13741c76-16a1-11e6-81f5-0b14b6f77d70.png)

**Comment by jboz62** [comment]
Ok, thanks for the heads up. Where did this contact form come from? I don’t see a contact form on other sites, did Steve add this?

**Comment by meetinghouse** [comment]
Funny, I assumed this was something you created as you've been working with
him.

I bet he copied the HTML over from his old site,
http://www.newenglandsoundproofing.com/contact_form.html

**Comment by jboz62** [comment]
Yep, that’s it, he’s been working on the site some, I noticed. He made a lot of changes to the menu order too, which is good, that’s why we created that menu system :)

Can you find out something more about the form, what controls it, etc. I’ll check the source code to see if there are any scripts that control it.

Thank you.

**Comment by KathJohnson** [comment]
I'm suggesting that he move to a simple list for his contact page for now. We can replace later with a form, or add a form.

**Comment by jboz62** [comment]
Yes, that’s probably a good idea. I would also ask Sudhir if he has any ideas on this too. 

If he wants to use a form service, that might be one way to go. We could just embed the form, that way. I was playing around with the CKEditor package and downloaded some custom builds I put together. The newer version does have some form fields on there, but not sure how well that works or how easy it is to use. 

I’d like to upgrade that CKEditor soon so we have more tools to use and more functionality for clients, like the styles dropdown we discussed and the captioned images feature. It’s just a matter choosing the features and downloading the package. Then we would just commit the new editor folder to the repository after testing locally. http://ckeditor.com/download <http://ckeditor.com/download> (on this link you can choose one of the packages and then click the customize radio button to add to it).

**Comment by 1sp** [comment]
if it is a simple contact-us form, another approach could be that we give them an embeddable form as a premade token. If that token is present in the body the system would insert a form there

e.g token is [contact-form] . Client creates a page and writes some text in the page's body and at the end writes [contact-form]. When saved, this page will have an embedded form (along with the other text he wrote)

**Comment by 1sp** [comment]
Actually this approach could be used for embedding any interactive control. The form would have its own css hence would be themable

**Comment by jboz62** [comment]
That’s similar to a shortcode in WordPress, good idea. How would you ad the form action, where to send the form? His old site had some kind of service URL in the form action. Looks like this site is live now, so I don’t have the old URL.

**Comment by 1sp** [comment]
Yes, even drupal use a similar system.
I assumed he just want the form submissions in the email. In case he need an ability to add custom action url for the form then action url can be put inside the shortcode e.g [contact-form:http://example.com/post]

Another option could be to expose custom setting for providing form action url

**Comment by jboz62** [comment]
That sounds good. How much time would it take to get this done, Sudhir?

Katherine, any thoughts on this? Might be something to ask Steve about.

**Comment by 1sp** [comment]
John, this feature would take around 3 hours (form action url passed with the shortcode)

**Comment by KathJohnson** [comment]
Let's do this. 

Also, it seems we should upgrade the ckeditor to the new version.

**Comment by jboz62** [comment]
@1sp Sudhir, did you get that update, Katherine wants to go ahead with the contact form setup for: http://www.newenglandsoundproofing.com/contact

@KathJohnson On the ckeditor, we should discuss this as you need to do a custom build of the features you want in the editor, e.g. Styles Dropdown, Image Captions, etc. If you go to this URL: http://ckeditor.com/download you will see they have 3 packages you can download, OR you can do a custom build. We may not want clients having some features, so it may be better to start with the Standard version and add plugins to it. You can decide on the approach, please review that and see what you think.
![screen shot 2016-07-13 at 2 43 47 pm](https://cloud.githubusercontent.com/assets/11739120/16815604/734917f2-4908-11e6-94ef-65b5ab70c088.png)

**Comment by jboz62** [comment]
@KathJohnson I guess that contact form setup is for all the sites, not just NES, correct?

**Comment by meetinghouse** [comment]
As an option, yes. That is possible, right? The contact form wouldn't exist
automatically already, but could be added.

**Comment by meetinghouse** [comment]
That sounds good.

>

**Comment by KathJohnson** [comment]
For the moment, we have set NES up with a form that live on Jotform.com. 

We have a free account there. It allows only up to 100 submissions per month (in other words, 100 e-mails), so this is probably not a good long term solution. It is a stopgap for now until we can program the ability to embed a simple pre-built contact form as described above.

**Comment by jboz62** [comment]
If Steve wants the CAPTCHA on this contact form, that's available in the settings on JotForm, it can be enabled.

**Comment by jboz62** [comment]
@meetinghouse John C., I changed the title of this issue and the labels to better reflect the direction on this issue. There are comments on this thread from Sudhir on the direction using tokens/shortcodes as a way to insert a form into the contact page. 

I've also added a new label called **"Enhancement for Client"** to use for items like this that are special requests. This would improve the base code, but you could bill a portion to Steve as he requested the functionality.

@vivek-chaudhari Please review this thread for future development and let me know your thoughts on this plan. This is a lower priority item, but may be addressed soon.

Thanks,
John

**Comment by vivek-chaudhari** [comment]
@jboz62 John, it is not easy to provide shortcode functionality as per sudhir's suggestion, because when we add it into editor, editor will adds its own html characters around shortcode so it will be difficult to get exact token/shortcode from contents.

**Comment by jboz62** [comment]
Vivek, we can edit the config file to allow the tokens. I did that to allow script tags and other css. You can edit the config file to allow for that, also. There are plugins that will allow for tokens too: http://ckeditor.com/addon/token <http://ckeditor.com/addon/token>

**Comment by jboz62** [comment]
@vivek-chaudhari  Vivek, here's the original placeholder/token plugin: http://ckeditor.com/addon/placeholder Not sure which is better.

I'm working on a new version of CKEditor here: http://testdeploy.corbettresearchgroup.com/login We're adding a lot of new functionality, let me know if you have any ideas. This has not been added to the repo or any other sites. I added it directly and will remove it once we deploy it.

**Comment by jboz62** [comment]
Vivek, did you get a chance to look this over and see if it will work? Thoughts?

**Comment by vivek-chaudhari** [comment]
I have not get chance to look at it. Tomorrow I will check it out.

**Comment by vivek-chaudhari** [comment]
@jboz62 this  http://testdeploy.corbettresearchgroup.com/login redirects me to https://restorationtrades.wordpress.com/wp-login.php url. I have added this plugin on my localhost and it allows us to insert token, but issue will arise when we need to replace token with original content when page loads contents. We need to save token value to somewhere.
According to me using token plugin is more better instead of placeholder plugin, because token may be predefined in CKEditor config files.

**Comment by jboz62** [comment]
@vivek-chaudhari It seems to be working now, can you clear your cache and try it again?

On the token plugin, you're saying that this plugin works best? http://ckeditor.com/addon/token

After adding the plugin, where is the contact form defined?

If you can give me some idea of what your development plan is and amount of time, we can get John Corbett's approval on this.

**Comment by vivek-chaudhari** [comment]
@jboz62  Now it working for me. It seems that you have not added token addon on ckeditor yet.
My development plan is.
- Create a new database table for tokens.
- Allow user to add token from backend( user can add, edit or delete token or inactive token). Have token_name=>token_value.
- After adding token, user will place token in content editor.
- whenever content is going to load, system needs to check for token. If have token, search for token value and replace token with that value.
- Token value will be anything(HTML,Iframe, JBContact Form Script) etc.
 If we develop like this. It will consume approx. 12-16 hours. If you have any other and simple idea, please suggest that.

**Comment by jboz62** [comment]
Thanks, Vivek. Let me review this and discuss with John C.

**Comment by jboz62** [comment]
@vivek-chaudhari Is it possible to do a setup like [this](http://wiyono05.blogspot.com/2014/02/create-simple-contact-form-in-laravel-4.html) or [this](http://stackoverflow.com/questions/20395884/contact-form-laravel-4) that would create a basic contact form on the contact page, but still allow for adding additional content on the page? Maybe something like that would work instead of the token idea? If you have any other suggestion, please advise.

**Comment by vivek-chaudhari** [comment]
@jboz62  Yes it is possible to develop like this and it will allowed to add any kind of contents. If we don't want to use tokens then we can add on additional plain textarea below content editor which will allow to add any kind of contents. And these additional content will display after editor contents.
